utils/ort/src/main/kotlin/UserInfoAuthenticator.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (See <https://github.com/oss-review-toolkit/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.ort
+
+import java.net.Authenticator
+import java.net.PasswordAuthentication
+
+/**
+ * A trivial [Authenticator] that checks if the credentials are already present in the URL. If this is the case, the
+ * credentials will be used for password authentication.
+ */
+class UserInfoAuthenticator : Authenticator() {
+    override fun getPasswordAuthentication(): PasswordAuthentication? {
+        if (requestorType == RequestorType.SERVER) {
+            requestingURL?.userInfo?.let {
+                val credentials = it.split(":")
```

Comment:
Nit: We could use the `limit = 2` argument to stop splitting after two parts.

Thinking about it, we probably even must do that to not omit any parts from the password (Which might contain unescaped colons?) below when using `credentials[1]`.